internal/server/validation.go

```diff
@@ -28,6 +30,20 @@ func (s *Server) Validate() error {
 	err := valid.Struct(s)
 	valerrs, ok := err.(validator.ValidationErrors)
 
+	if s.CustomNotFoundPage != "" {
+		if _, err := os.Stat(s.CustomNotFoundPage); err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				return fmt.Errorf("custom not found file %q does not exist", s.CustomNotFoundPage)
+			}
+
+			return err
+		}
+	}
+
+	if !(s.CustomNotFoundStatusCode >= 100 && s.CustomNotFoundStatusCode <= 511) {
+		return fmt.Errorf(" invalid custom not found status code: %d", s.CustomNotFoundStatusCode)
+	}
```

Comment:
I would probably cheat here, if we call `http.StatusText()` with the status code, an empty string returned means the status code is invalid.

```suggestion
	if s := http.StatusText(s.CustomNotFoundStatusCode); s == "" {
		return fmt.Errorf("invalid custom not found status code: %d", s.CustomNotFoundStatusCode)
	}
```